src/backend/distributed/utils/metadata_cache.c

```diff
@@ -1181,13 +1184,17 @@ CheckCitusVersion(int elevel)
  * this function logs an error with the specified elevel and returns false,
  * otherwise it returns true.
  */
-static bool
+bool
 CheckAvailableVersion(int elevel)
 {
 	char *availableVersion = NULL;
 
-	Assert(CitusHasBeenLoaded());
-	Assert(EnableVersionChecks);
+	if (citusVersionKnownCompatible ||
```

Comment:
We perform this check before PostgreSQL start to process ALTER EXTENSION. So creating_extension will be false at that moment and we will not incorrectly exit early.